pkgs/data/icons/beauty-line-icon-theme/default.nix

```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, gtk3, gnome-icon-theme, hicolor-icon-theme }:
+
+stdenv.mkDerivation rec {
+  pname   = "BeautyLine";
+  version = "0.0.1";
+
+  src = builtins.fetchTarball {
+    name   = "${pname}-${version}";
+    url    = "https://github.com/gvolpe/BeautyLine/releases/download/${version}/BeautyLine.tar.gz";
+    sha256 = "0s9zpxh8s93awnkw30xnxmpnqsw6wx1fvyvjh33fs3r40vv7lxgg";
+  };
+
+  nativeBuildInputs = [ gtk3 ];
+
+  propagatedBuildInputs = [ gnome-icon-theme hicolor-icon-theme ];
+
+  dontDropIconThemeCache = true;
+
+  installPhase = ''
+    mkdir -p $out/share/icons/${pname}
+    cp -r * $out/share/icons/${pname}/
+    gtk-update-icon-cache $out/share/icons/${pname} || true
```

Comment:
```suggestion
    gtk-update-icon-cache $out/share/icons/${pname}
```

Reply:
It fails without `|| true`, not sure if the `gtk-update-icon-cache` should be required.

Reply:
Why does it fail? It shouldn't fail and we don't want to hide the failure. It might be that you need to add a -f.

Reply:
I get this error

```
gtk-update-icon-cache: The generated cache was invalid.
```

I'm not really familiar with GTK so I don't know more. Adding `-f` did not help.

Reply:
This might have some answers, I'll see if this can be properly fixed: https://askubuntu.com/questions/1080257/gtk-update-icon-cache-the-generated-cache-was-invalid

Reply:
I tried both suggestions in that thread and nothing worked, unfortunately. I left a question on the [official site](https://www.gnome-look.org/p/1425426/) but I doubt I'll get a response anytime soon.

FWIW I'm using this icon theme on my machine. If anyone out there wants to give it a try now, you can have a look here (using Home Manager): https://github.com/gvolpe/nix-config/blob/master/home/themes/default.nix

Reply:
The following file is the culprit:
```
BeautyLine/apps/scalable/goa-account-msn.svg'$'\n''.svg'
```
Rename it to remove the trailing garbage `'$'\n''.svg'` and `gtk-update-icon-cache` should work.

I would also suggest to open an issue upstream to fix it.

Reply:
Thanks @romildo ! I'll try that. Unfortunately, there's no upstream at all :laughing: that's why I uploaded it to my own Github repo.